### PR TITLE
Cld 233 simplify database

### PR DESCRIPTION
**Pull request by davidmdm**

### CODE REVIEW

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

File: src/libs/repositories/item.repository.ts [hunk 1]
```diff
@@ -110,46 +76,51 @@ export class ItemRepository {
...
+        .whereIn(itemRelTable.rel_item_id, uniq(relationshipsToCreate.map(x => x.rel_item_id)))
+        .withSchema(accountId);
+
       for (const rel of relationshipsToCreate) {
-        await upsert(trx, {
-          insert: trx
+        const previousReference = !!previousRels.find(
+          x => x.item_id === rel.item_id && x.rel_item_id === rel.rel_item_id
+        );
+        if (previousReference) {
```
Comment (mlaflamm) [inline src/libs/repositories/item.repository.ts]: We need some kind of comment of what is going on here and also for previous parent

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

**Comment by mlaflamm** [comment]
`E2_CLIENT_ID` and `E2_CLIENT_SECRET` are missing from the `secret.yaml` template file

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

**Comment by mlaflamm** [comment]
> `E2_CLIENT_ID` and `E2_CLIENT_SECRET` are missing from the `secret.yaml` template file

Done

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

File: src/libs/repositories/item.repository.ts [hunk 2]
```diff
@@ -71,70 +70,84 @@ export class ItemRepository {
...
+            if (rowCount === 0) {
+              throw httpError(500, `Failed to update relationship: ${JSON.stringify(itemRel)}`);
+            }
+            continue;
+          }
+          await trx
+            .withSchema(accountId)
+            .insert(itemRel)
+            .into(itemRelTable.tableName());
+        }
```
Comment (mlaflamm) [inline src/libs/repositories/item.repository.ts]: This block of code is insane! This is so long that we forget we are in a loop. There are 4 code branches split by `else` or `continue`!!!. If someone want to add code after this block, these `continue` folly are sure to make drive him mad! Simply extract this block in a separate function and use `return`!

Comment (mlaflamm) [reply src/libs/repositories/item.repository.ts]: And comment each branch please!

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-index-pr-4** [here](http://e2-index.jx-lvlstudio-e2-index-pr-4.e2-dev.com)

**Pull request merged**